app/models/pubmed_source_record.rb

```diff
@@ -123,7 +123,19 @@ def source_as_hash(publication = Nokogiri::XML(source_data).xpath('//PubmedArtic
     record_as_hash[:author] = author_list.map { |a| author_to_hash(a) }.compact
 
     record_as_hash[:mesh_headings] = mesh_headings unless mesh_headings.blank?
-    record_as_hash[:year] = publication.xpath('MedlineCitation/Article/Journal/JournalIssue/PubDate/Year').text unless publication.xpath('MedlineCitation/Article/Journal/JournalIssue/PubDate/Year').blank?
+    year_xpaths = [
+      'MedlineCitation/Article/Journal/JournalIssue/PubDate/Year',
+      'MedlineCitation/Article/ArticleDate/Year',
+      'PubmedData/History/PubMedPubDate[@PubStatus="accepted"]/Year'
+    ]
+    # look for a year in all of the xpath locations above in order
+    #  stop after the first produces something that looks like a year
+    year_xpaths.each do |path|
+      if !publication.xpath(path).blank? && publication.xpath(path).text.match(/[12][0-9]{3}/)
```

Comment:
We can convert this to reuse the initial regex match though, like:
```ruby
next if publication.xpath(path).blank?
match = publication.xpath(path).text.match(/[12][0-9]{3}/)
next unless match
record_as_hash[:year] = match.to_s
break
```

Or more cleanly, we don't need to `break` the loop explicitly if we rely on how Nokogiri works and use `find`, like:
```ruby
match = year_xpaths.map { publication.xpath(path).text }.find { |x| x.match(/[12][0-9]{3}/}
```
We can count on a `blank?` Document node to still return `''` to `.text`, neither breaking nor matching the regex.  Example:
```ruby
> Nokogiri::XML::Document.new.xpath('foo').text
=> ""
```
I think _some_ form of not regex-ing twice seems useful.

Reply:
I think your strategy works -- see what I just pushed (your original suggestion just had some typos and missing characters)